File: frontend/src/api/patients.js
// frontend/src/api/patients.js

import api from "./index";

export const getPatients = async () => {
  try {
    console.log("📡 Fetching patients from API...");
    const response = await api.get("/patients/");
    console.log("✅ Patients fetched:", response.data);
    return response.data;
  } catch (error) {
    console.error("❌ Get patients error:", error);
    throw error;
  }
};

export const getPatient = async (id) => {
  try {
    const response = await api.get(`/patients/${id}`);
    return response.data;
  } catch (error) {
    console.error("❌ Get patient error:", error);
    throw error;
  }
};

export const createPatient = async (data) => {
  try {
    console.log("📡 Creating patient with data:", data);
    const response = await api.post("/patients/", data);
    console.log("✅ Patient created:", response.data);
    return response.data;
  } catch (error) {
    console.error("❌ Create patient error:", error.response?.data || error);
    throw error;
  }
};

export const updatePatient = async (id, data) => {
  try {
    console.log("📡 Updating patient with ID:", id);
    console.log("📡 Update data:", data);
    const response = await api.put(`/patients/${id}`, data);
    console.log("✅ Patient updated:", response.data);
    return response.data;
  } catch (error) {
    console.error("❌ Update patient error:", error.response?.data || error);
    throw error;
  }
};

export const deletePatient = async (id) => {
  try {
    console.log("📡 Deleting patient with ID:", id);
    const response = await api.delete(`/patients/${id}`);
    console.log("✅ Patient deleted");
    return response.data;
  } catch (error) {
    console.error("❌ Delete patient error:", error.response?.data || error);
    throw error;
  }
};

File: backend/app/api/v1/patients.py
from fastapi import APIRouter, Depends, HTTPException, Query, status
from sqlalchemy.orm import Session
from typing import List, Optional
from ...core.database import get_db
from ...services.patient_service import PatientService
from ...schemas.patient import PatientCreate, PatientUpdate, PatientOut
from ...core.security import get_current_user
import uuid

router = APIRouter(prefix="/patients", tags=["Patients"])

@router.get("/", response_model=List[PatientOut])
def get_patients(
    skip: int = Query(0, ge=0),
    limit: int = Query(100, ge=1, le=100),
    search: Optional[str] = None,
    db: Session = Depends(get_db),
    current_user = Depends(get_current_user)
):
    service = PatientService(db)
    return service.get_all(skip, limit, search)

@router.get("/{patient_id}", response_model=PatientOut)
def get_patient(
    patient_id: uuid.UUID,
    db: Session = Depends(get_db),
    current_user = Depends(get_current_user)
):
    service = PatientService(db)
    patient = service.get_by_id(patient_id)
    if not patient:
        raise HTTPException(status_code=404, detail="Patient not found")
    return patient

@router.post("/", response_model=PatientOut, status_code=status.HTTP_201_CREATED)
def create_patient(
    data: PatientCreate,
    db: Session = Depends(get_db),
    current_user = Depends(get_current_user)
):
    service = PatientService(db)
    try:
        return service.create(data)
    except Exception as e:
        raise HTTPException(status_code=400, detail=str(e))

@router.put("/{patient_id}", response_model=PatientOut)
def update_patient(
    patient_id: uuid.UUID,
    data: PatientUpdate,
    db: Session = Depends(get_db),
    current_user = Depends(get_current_user)
):
    service = PatientService(db)
    patient = service.update(patient_id, data)
    if not patient:
        raise HTTPException(status_code=404, detail="Patient not found")
    return patient

@router.delete("/{patient_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_patient(
    patient_id: uuid.UUID,
    db: Session = Depends(get_db),
    current_user = Depends(get_current_user)
):
    service = PatientService(db)
    if not service.delete(patient_id):
        raise HTTPException(status_code=404, detail="Patient not found")
    return None